components/onboarding/CategorySelection.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+    <div class="h-screen w-full bg-primary-bg p-2 pt-6 overflow-hidden">
+        <div class="flex flex-col">
+            <!-- Header text -->
+            <h1 class="landscape:text-4xl landscape:my-12 landscape:mb-20 text-4xl font-bold text-primary-text mb-8 text-center">
+                {{ t('onboarding.selectcategory') }}
+            </h1>
+            <!-- Category selection grid -->
+            <div
+                class="grid grid-cols-[.18fr,.18fr,.18fr] gap-x-2 gap-y-4 mx-1 justify-content-center
+                    portrait:mb-8 landscape:mb-8"
+            >
+                <!-- Category buttons -->
+                <button
+                    v-for="category in categories"
+                    :key="category.code"
+                    type="button"
+                    class="bg-secondary-bg/90 hover:bg-secondary-bg transition-all
+                        rounded-xl pt-3 pb-4 px-2 text-center shadow hover:shadow-xl
+                        transform hover:scale-105 w-24 landscape:w-64 whitespace-pre-line text-sm"
+                    @click="selectCategory(category.code)"
+                >
+                    <div class="flex justify-center items-start">
+                        <span class="inline-block w-8 h-8">
+                            <component
+                                :is="categoryIconMap[category.code]"
+                                class="w-8 h-8 fill-currentColor"
+                                aria-hidden="true"
+                            />
+                        </span>
+                    </div>
+                    <span class="relative font-medium text-primary-text">
+                        {{ category.displayText }}
+                    </span>
+                </button>
+            </div>
+            <!-- "Something else" button -->
```

Comment:
Nit: This could be a name or ID instead of a comment.

Reply:
honestly, I never use name but maybe I should more 🤔  It still has the downside of visibility when scrolling through a lot of html content. 
> Comments are generally more helpful if they're used to explain a decision or how something works if it's not 
> immediately obvious.

I love that you're using my own words against my comment 😆 In Javascript, I totally agree and that is my philosophy. Only reason I'm pitching here is because you can't easily use functions or variables in html for better visibility. If there was a standardized way to highlight the name field, it would eliminate this problem 🤔